x-pack/plugins/fleet/dev_docs/local_setup/enrolling_agents.md

```diff
@@ -0,0 +1,228 @@
+
+## Kibana config
+
+Add the following to your `kibana.dev.yml`. Note that the only differences between VM and container setups are the URLs of the Fleet Server host and Elasticsearch output. If you want to set up a Fleet Server on a VM, you will first need to launch the VM in order to get the IP address.
+
+```yml
+# Set the Kibana server address to Fleet Server default host.
+server.host: 0.0.0.0
+
+# Install Fleet Server package.
+xpack.fleet.packages:
+  - name: fleet_server
+    version: latest
+
+# Create an agent policy for Fleet Server.
+xpack.fleet.agentPolicies:
+  - name: Fleet Server policy
+    id: fleet-server-policy
+    is_default_fleet_server: true
+    # is_managed: true # Useful to mimic cloud environment
+    description: Fleet server policy
+    namespace: default
+    package_policies:
+      - name: Fleet Server
+        package:
+          name: fleet_server
+        inputs:
+          - type: fleet-server
+            keep_enabled: true
+            vars:
+              - name: host
+                value: 0.0.0.0
+                frozen: true
+              - name: port
+                value: 8220
+                frozen: true
+
+# Set a default Fleet Server host.
+xpack.fleet.fleetServerHosts:
+  - id: default-fleet-server
+    name: Default Fleet server
+    is_default: true
+    # host_urls: [https://<the-IP-address-of-your-VM>:8220] # For running a Fleet Server in a VM
+    # host_urls: ['https://host.docker.internal:8220'] # For running a Fleet Server Docker container
+
+# Set a default Elasticsearch output.
+xpack.fleet.outputs:
+  - id: es-default-output
+    name: Default output
+    type: elasticsearch
+    is_default: true
+    is_default_monitoring: true
+    # hosts: ['http://<your-local-IP>:9200'] # For enrolling agents on VM
+    # hosts: ['http://host.docker.internal:9200'] # For enrolling dockerized agents
+```
+
+## Using Multipass VMs
+
+[Multipass](https://multipass.run/) is lightweight virtualization tool for running Ubuntu VMs. Follow instructions at https://multipass.run/install or to install Multipass on your local machine.
+
+Advantages of running Elastic Agents on a VM include:
+- More realistic setup.
+- Ability to use the `elastic-agent` commands, e.g. `sudo elastic-agent status`, `sudo elastic-agent restart`...
+- Agents can be upgraded.
+- Elastic Defend can be installed.
+
+To run a Fleet Server and agents on VMs, make sure the default output host defined in your `kibana.dev.yml` uses your local IP (NB: using `localhost` can cause connection issues). The default Fleet Server host can be set once the VM for the Fleet Server is running. In Fleet UI, this should be reflected in Fleet settings:
```

Comment:
quite silly but can you include how to retrieve the correct local IP to use?

Reply:
We can probably provide the Mac way which is 

```
$ ipconfig getifaddr en0
```

Reply:
No such thing as silly, this is the kind of stuff that can save someone time 🙂 I usually Alt+click on my wifi dropdown menu in the Mac top bar, but +1 to the command line way 👍